clconf/template.go

```diff
@@ -0,0 +1,198 @@
+package clconf
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+const setuid = 4
+const setgid = 2
+const sticky = 1
+
+// Settings for ProcessTemplates.
+type TemplateSettings struct {
+	Flatten           bool
+	KeepExistingPerms bool
+	Rm                bool
+	FileMode          string
```

Comment:
At an API level (which `clconf` package is), `FileMode` should be of type `os.FileMode`.  The string here is due to CLI processing and thus belongs in `cmd`.

Reply:
While I can do this for `DirMode`, `FileMode` is optional and since an `os.FileMode` is really just an integer there'd be no way to distinguish between populated and not. If I'm not doing it for `FileMode` can you think of a reason to do it for `DirMode`?

Reply:
what do you think of something like the [optional](https://github.com/googleapis/google-cloud-go/blob/master/internal/optional/optional.go) approach?  I'm not sure i like it, but also not sure i like using string here...  Up to you...

Reply:
I feel like in Java where that's even part of the language it has issues. If it were a golang built-in I'd say sure, but tacking on a library for it feels hacky. Given something as complicated as a path is a string I don't really buy the argument that "objects are preferred" is idiomatic.